torch/testing/_internal/common_utils.py

```diff
@@ -571,9 +571,22 @@ def run_tests(argv=UNITTEST_ARGS):
     # Determine the test launch mechanism
     if TEST_DISCOVER:
         _print_test_names()
-    elif TEST_IN_SUBPROCESS:
-        suite = unittest.TestLoader().loadTestsFromModule(__main__)
-        test_cases = discover_test_cases_recursively(suite)
+        return
+
+    # Before running the tests, lint to check that every test case extends from TestCase
+    suite = unittest.TestLoader().loadTestsFromModule(__main__)
+    test_cases = discover_test_cases_recursively(suite)
+    failed = False
+    for case in test_cases:
```

Comment:
How long does it take to run this check? Can we perhaps speed it up, by making `discover_test_cases_recursively` to return test_suites (aka classes), rather than cases?

Reply:
Resolved in a different way that should take shorter